app/src/main/java/com/lucas/medicaltools/MainActivity.kt

```diff
@@ -1,11 +1,107 @@
 package com.lucas.medicaltools
 
-import androidx.appcompat.app.AppCompatActivity
+import android.content.Intent
 import android.os.Bundle
+import android.view.View
+import android.widget.Toast
+import androidx.appcompat.widget.SearchView
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.hannesdorfmann.mosby3.mvi.MviActivity
+import com.jakewharton.rxrelay2.PublishRelay
+import com.jakewharton.rxrelay2.Relay
+import com.lucas.medical_equip.repository.MedicalTool
+import com.lucas.medicaltools.databinding.ActivityMainBinding
+import io.reactivex.subjects.PublishSubject
+import mu.KotlinLogging
+import timber.log.Timber
 
-class MainActivity : AppCompatActivity() {
+private val logger = KotlinLogging.logger {}
+class MainActivity : MviActivity<MainView, MainPresenter>(), MainView {
+
+    private lateinit var medicalTools: List<MedicalTool>
+    private lateinit var medicalAdapter: MedicalToolAdapter
+    private lateinit var searchView: SearchView
+    private lateinit var binding: ActivityMainBinding
+    private val onCreateHappenedSubject = PublishSubject.create<String>()
+    private val useCapturedFilteringTextSubject = PublishSubject.create<String>()
+    private val onScreenLoadRelay = PublishRelay.create<Intent>()
+    private var filterKey: String = ""
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_main)
+        Timber.d("ACTIVITY: IN ON CREATE")
+        binding = ActivityMainBinding.inflate(layoutInflater)
+        setContentView(binding.root)
```

Comment:
call your on create publish subject so that the flow of MVI is kicked off and you actually fetch data.